tests/api/worlds/nested_def_nodes.wbt

```diff
@@ -0,0 +1,129 @@
+#VRML_SIM R2020a utf8
+WorldInfo {
+  basicTimeStep 16
+}
+Viewpoint {
+  fieldOfView 0.85
+  orientation -0.08888575959716818 0.955941011161537 0.27977867131054257 6.26114482885042
+  position 0.17053014674645414 0.1607735089540114 2.3415875442581355
+}
+Background {
+  skyColor [
+    1 1 1
+  ]
+}
+PointLight {
+  attenuation 0 0 1
+  intensity 10
+  location -4 6 5
+  castShadows TRUE
+}
+DEF USE_CASE_1 Solid {
+  children [
+    DEF G Group {
+      children [
+        DEF T1 Transform {
+          children [
+            DEF S1 Shape {
+              appearance Appearance {
+                material Material {
+                  diffuseColor 0.305882 0.603922 0.0235294
+                }
+              }
+              geometry DEF B1 Box {
+                size 0.1 0.1 0.1
+              }
+            }
+          ]
+        }
+        DEF T2 Transform {
+          translation 0.4 0 0
+          children [
+            USE S1
+          ]
+        }
```

Comment:
Adding a `USE T2` here causes an issue:
  - changing the `B1` size works fine.
  - changing the `T2` translation works fine for the shape but not for the bounding object (a save and revert is still required for the update to be effective).

Reply:
I added a test for this case below (`USE_CASE_4`): test the update of a USE node containing another USE node.